core/commands/add.go

```diff
@@ -103,13 +108,24 @@ remains to be implemented.
 		hidden, _, _ := req.Option(hiddenOptionName).Bool()
 		chunker, _, _ := req.Option(chunkerOptionName).String()
 
+		e := dagutils.NewDagEditor(n.DAG, newDirNode())
 		if hash {
-			nilnode, err := core.NewNodeBuilder().Build(n.Context())
+			nilnode, err := core.NewNodeBuilder().NilRepo().Build(n.Context())
 			if err != nil {
 				res.SetError(err, cmds.ErrNormal)
 				return
 			}
 			n = nilnode
+
+			// build mem-datastore for editor's intermediary nodes
+			bs := bstore.NewBlockstore(syncds.MutexWrap(ds.NewMapDatastore()))
+			bsrv, err := bserv.New(bs, offline.Exchange(bs))
+			if err != nil {
+				res.SetError(err, cmds.ErrNormal)
+				return
+			}
+			memds := dag.NewDAGService(bsrv)
```

Comment:
can this be moved out of here? all this setup doesnt belong in this command. 
- i think this will already be what happens when the node is created without `NilRepo()` (i.e. defaults to `NewMapDatastore()`)
- if not, maybe move this stuff somewhere else so this command can be clean like:

```
eds := n.DAG
if hash {
  nn, err := NilNode()
  if err != nil {
    res.SetError(err, cmds.ErrNormal)
    return
  }
  n = nn
  memds, err := NewMemDagService()
  if err != nil {
    res.SetError(err, cmds.ErrNormal)
    return
  }
  eds = memds
}
e := dagutils.NewDagEditor(eds, newDirNode())
```